### DISCUSSION TITLE
How to glob with Nelua during compilation

### DISCUSSION THREAD

**Discussion by stefanos82** [edubart/nelua-lang#82] [Q&A] [answered]
I have a folder full of Nelua files; what I tend to do is to execute the following command in terminal: `for name in *.nelua; do nelua "$name"; done`.

It's not that difficult, but I have tried the shortest command `nelua *.nelua` and expected it to work; to my surprise it didn't and I wondered why.

Can you enlighten me please?

**Comment by edubart** [comment, marked as answer]
This is not possible because arguments following the input file are reserved to be arguments to be passed to the running application, example:

```lua
require 'arg'
print(arg[1])
```

Then running:
```bash
$ nelua test.nelua hello.nelua
hello.nelua
```

And using `*.nelua` is effectively the same as expanding current directory files into many arguments.

**Reply by stefanos82** [reply]
Ah OK; then my standard way with `for` loop will be just fine.

Thank you for letting me know.
